specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -1,26 +1,89 @@
-## Messaging Systems
+# Messaging Systems
 
 The instrumentation of messaging systems includes:
-- Transaction creation on message reception, either as a child of the message-sending span, if the sending action is traced (meaning 
+
+- Transaction creation for received message processing, either as a child 
+of the message sending span if the sending operation is traced (meaning 
 distributed tracing support), or as root
-- Span creation for message sending action
-- Span creation on message *polling that occurs within a traced transaction*
+- Span creation for message sending operation
+- Span creation for message *receiving operation that occurs within a traced transaction*
+
+## Message sending/publishing
+
+A Message send/publish SHOULD be captured as a `messaging` span only if 
+occurring within a traced transaction.
+
+![publish](uml/publish.svg)
+
+### Trace Context
+
+if the messaging system exposes an API for sending additional message properties/metadata, it
+SHOULD be used to propagate the [Trace Context](https://www.w3.org/TR/trace-context/) of the
+`messaging` span, to continue the [distributed trace](tracing-distributed-tracing.md).
+
+## Message reception/consumption
+
+Message reception can be divided into two types, **passive** and **active**. 
+
+### Passive message reception
+
+Passive message reception typically involves a listener or callback subscribed
+to a queue/topic/subscription that is called once a message or batch of messages
+is available, and passed the message or batch of messages for processing.
+
+Passive reception SHOULD be captured as a `messaging` transaction that starts when
+a message or batch of messages is received for processing, and ends when the
+processing of the message or batch of messages finishes.
+
+TODO: insert Azure Service Bus example
```

Comment:
Was this a TODO for yourself that you missed?

Reply:
Yeah 🙂 I'll look at adding a diagram

Reply:
I don't know that it's strictly necessary. But I would remove the TODO either way.

Reply:
Agreed. I'll remove for now and look to revisit when I get a chance